tests/test_contexts.py

```diff
@@ -46,6 +47,26 @@ def exit(self):
     self.depth = None
 
 
+@contexts.register
+class AsyncSimpleContext(contexts.AsyncContext):
+  """An async context object for testing."""
+
+  def __init__(self):
+    super().__init__()
+    self.depth: Optional[int] = None
+
+  async def aenter(self):
```

Comment:
super nit: `assert self.depth is None` ?

At the moment it looks like this is almost re-entrant but `__exit__()` would cause things to blow up by setting depth to `None` instead of only setting `depth` to `None` if it's `1`.

Reply:
Done